scipy/linalg/_solvers.py

```diff
@@ -121,7 +123,53 @@ def solve_lyapunov(a, q):
 
     """
 
-    return solve_sylvester(a, a.conj().transpose(), q)
+    a = np.atleast_2d(_asarray_validated(a, check_finite=True))
+    q = np.atleast_2d(_asarray_validated(q, check_finite=True))
+
+    r_or_c = float
+
+    for ind, _ in enumerate((a, q)):
+        if np.iscomplexobj(_):
+            r_or_c = complex
+
+        if not np.equal(*_.shape):
+            raise ValueError("Matrix {} should be square.".format("aq"[ind]))
+
+    # Shape consistency check
+    if a.shape != q.shape:
+        raise ValueError("Matrix a and q should have the same shape.")
+
+    # Compute the Schur decomp form of a
+    r, u = schur(a, output='real')
+
+    # Construct f = u'*q*u
+    f = u.conj().T.dot(q.dot(u))
+
+    # Call the Sylvester equation solver
+    trsyl, = get_lapack_funcs(('trsyl',), (r, f))
+    if trsyl is None:
```

Comment:
I think this check is not needed --- `get_lapack_funcs` raises if it fails to find.
You should be able to also write `trsyl = get_lapack_funcs('trsyl', (r, f))`